repo.go

```diff
@@ -57,6 +58,87 @@ func newRepo(repo *git.Repository, auth transport.AuthMethod) *Repo {
 	}
 }
 
+func cleanNewRepo(repo *git.Repository) error {
+	err := checkoutHeadHash(repo)
+	if err != nil {
+		return fmt.Errorf("error checking out HEAD: %v", err)
+	}
+
+	err = cleanLocalBranches(repo)
+	if err != nil {
+		return fmt.Errorf("error cleaning local branches: %v", err)
+	}
+
+	err = cleanLocalReferences(repo)
+	if err != nil {
+		return fmt.Errorf("error cleaning local references: %v", err)
+	}
+	return nil
+}
+
+func checkoutHeadHash(repo *git.Repository) error {
+	head, err := repo.Head()
+	if err != nil {
+		return fmt.Errorf("unable to resolve HEAD commit: %v", err)
+	}
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return fmt.Errorf("unable to load worktree: %v", err)
+	}
+
+	err = worktree.Checkout(&git.CheckoutOptions{
+		Hash:  head.Hash(),
+		Force: true,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to checkout HEAD hash: %v", err)
+	}
+	return nil
+}
+
+func cleanLocalBranches(repo *git.Repository) error {
+	branches, err := repo.Branches()
+	if err != nil {
+		return fmt.Errorf("unable to load branches: %v", err)
+	}
+	err = branches.ForEach(func(ref *plumbing.Reference) error {
+		if ref.Name().IsBranch() {
+			// This is a locally checked out branch
```

Comment:
replace "checked out" with "stored". Checked out tends to refer to "the current working tree is X branch"